### ISSUE TITLE
Will wanawiki include the option to decrypt demo files (i.e. .wncyr)?

### ISSUE THREAD

**Issue by efialtes** [gentilkiwi/wanakiwi#22]
It seems that wanakiki does not support the possibility to decrypt demo files (.i.e. the ones with wncyr extension), right? Do you find it valuable to include such feature, to potentially cover 100% of the encrypted files?

According to http://blog.checkpoint.com/2017/05/14/wannacry-paid-time-off/
 there is a 'second encrypt/decrypt routine for a few files you can decrypt as a “free demo”. [...] They are also encrypted with a unique key per file. However, the private RSA key for these 10 is stored locally on the victim’s computer. [...] As can be seen in Figure A, the main decryption function tried to call a “.dky” file- supposedly containing the private RSA key, and with it decrypt all the files on the victims computer. In Figure B, we can see a similar function, this time calling the “f.wnry” file dropped by the encryptor module, which contains a list of the demo files. The private RSA key is already hard coded into the module. Both functions call the module import_RSA_key (fig. C)- the main decryptor with the path to the “.dky” file as an argument, and the demo with null instead.'

Also https://www.fireeye.com/blog/threat-research/2017/05/wannacry-malware-profile.html provides valuable info about demo files encrypt/decrypt.

**Comment by gentilkiwi** [comment]
It's 'kiwi', not wiki ;)
It's already supported in 0.2 (see releases). I already had it on wanadecrypt, so I pushed this on 0.2: https://github.com/gentilkiwi/wanakiwi/blob/master/wanakiwi/rsautil.c#L176

**Comment by efialtes** [comment]
Hi again!
I suggest a minor improvement, it looks like wanaKiwi (hope I did not misspelled again!) validates .wncry extensions so it does not accept .wncyr ones.

https://github.com/gentilkiwi/wanakiwi/blob/master/wanakiwi/wanakiwi.c#L281

Unfortunately I did't succeeded by changing manually the extension of a sample .wncyr file into a .wncry. 
When I executed 
wanakiwi.exe -search:<path1> -priv:<path2>00000000.dky -forcedecrypt

where 
path1: path for sample wncyr files renamed as wncry
path2: path to obtain a dummy 00000000.dky file, since I am testing only with sample wncyr files

The program shown the following error
Private key file on command-line: <path2>00000000.dky, will use it instead searching
Now searching .WNCRY files in <path1>\...
File <path1>\sample1.mp4.WNCRY -- ERROR rsautil_decryptFileWithKey ; CryptDecrypt(RSA) (0x00000057)
}

Not sure if this error is due to changing manually the .wncyr extension into .wncry; or the dummy 00000000.dky does not have a proper format :(

Thanks for your help!
